migrator/src/main/java/com/google/phonenumbers/migrator/MetadataReader.java

```diff
@@ -0,0 +1,74 @@
+package com.google.phonenumbers.migrator;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.i18n.phonenumbers.metadata.model.RangesTableSchema;
+import com.google.i18n.phonenumbers.metadata.table.CsvTable;
+import com.google.i18n.phonenumbers.metadata.table.RangeKey;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.Optional;
+import java.util.Scanner;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+/**
+ * A class used to represent a given metadata zip file. Each {@link MetadataReader} instance
+ * represents a zip file containing Csv Tables that can be imported based on specified
+ * region codes without the need to extract the whole zip file.
+ */
+public final class MetadataReader {
+  private final ZipFile metadataZipFile;
+
+  public MetadataReader(String fileLocation) throws IOException {
+    this.metadataZipFile = new ZipFile(fileLocation);
+  }
+
+  /**
+   * Returns an Optional {@link CsvTable} correlating to a given regionCode. If no ranges.csv file
```

Comment:
[Nit picking on docs] The great thing about Optional, is that using it makes the 2nd sentence here redundant. It's totally clear when empty would be returned (since it still has IOException in the signature for "real" errors). You could remove the second sentence and replace it with adding just "if present." at the end of the first sentence. You also don't need to say "an Optional ..." since that's clear from the type being used.

Great documentation is hard, and there's good value in being precise. This method returns "the CsvTable for a given region" , not "a CsvTable ...". There's only one thing we could ever return.

It's also doing "work" inside here (esp. since it can fail with IO exception, so maybe it needs a stronger verb than "get" (get implies "pick up something have and hand it back", which is not quite true in this case). Since the class is called a "reader", the natural name for the primary method would use the word "read" (or you could go with "Importer" and "import" since that's the naming convention of the code you're calling into, but that's your call).

Also consider using the "PhoneRegion" class rather than a String (Ctrl+N and start typing a class name in IntelliJ to find it). This just makes it more obvious what you're doing.